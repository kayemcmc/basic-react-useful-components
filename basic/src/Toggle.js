import React, { Component } from 'react';

export default class Toggle extends Component {
  state = {
    on: false
  };
  toggle = () => {
    this.setState({
      on: !this.state.on
    });
  };
  render() {
    return (
      <div>
        {this.state.on && <h1>G-wagon, G-wagon, G-wagon, G-wagon,G-wagon</h1>}
        <button onClick={this.toggle}>Show/Hide</button>
      </div>
    );
  }
}
