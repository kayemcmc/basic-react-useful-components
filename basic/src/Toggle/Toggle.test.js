import React from 'react';
import Toggle from './Toggle';
import { mount } from 'enzyme';
import * as renderer from 'react-test-renderer';

describe('Toggle', () => {
  // it('renders a label', () => {
  //     const component = mount(<Toggle label="Label" />);
  //     expect(
  //         component
  //             .find('.class')
  //             .first()
  //             .text()
  //     ).toEqual('Label');
  // });
  it('renders toggle correctly', () => {
    const component = renderer.create(<Toggle />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
