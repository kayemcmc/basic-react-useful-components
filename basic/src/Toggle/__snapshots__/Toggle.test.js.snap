// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle renders toggle correctly 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
